Field.php

```diff
@@ -163,28 +176,35 @@ public function start($_output)
     {
         $_output->outputField($this);
 
-        for ($i = 0; $i < $this->maxSteps; $i++) {
+        for ($i = 0; $i < $this->maxSteps; $i++)
+        {
             $this->history[] = $this->field; //saves field
             $this->generationCount++;
             $this->nextGeneration();
             $_output->outputField($this);
 
             //compare
-            foreach ($this->history as $previousField) {
+            foreach ($this->history as $previousField)
```

Comment:
To reduce code duplication please do the following:
1. Implement a method isEqualTo($_field) inside the Field class. This method should compare itself with the Field that is given as parameter to the method. It should return true if the fields are equal and false if not.
2. Implement a unit-test that tests that this method works correctly.
3. Use this method here to compare the fields here in this part of the code instead of the manual loop.
4. Use this method in all unit-tests where you need to compare fields.